src/validation.cpp

```diff
@@ -4600,7 +4607,14 @@ void Chainstate::LoadExternalBlockFile(
                 // Activate the genesis block so normal node progress can continue
                 if (hash == params.GetConsensus().hashGenesisBlock) {
                     BlockValidationState state;
-                    if (!ActivateBestChain(state, nullptr)) {
+                    bool genesis_activation_failure = false;
+                    for (auto c : GetAll()) {
+                        if (!c->ActivateBestChain(state, nullptr)) {
```

Comment:
In commit "Move block-storage-related logic to ChainstateManager" (034f920d1755cff752d46b089ad37bdd703bf896)

I think it would be a little clearer to move the `BlockValidationState state;` variable declaration down into the `for` loop so it is obvious the variable is ignored and not used to share state between `ActivateBestChain` calls